source/common/upstream/health_checker_base_impl.h

```diff
@@ -76,7 +76,8 @@ class HealthCheckerImplBase : public HealthChecker,
   class ActiveHealthCheckSession : public Event::DeferredDeletable {
   public:
     ~ActiveHealthCheckSession() override;
-    HealthTransition setUnhealthy(envoy::data::core::v3::HealthCheckFailureType type);
+    HealthTransition setUnhealthy(envoy::data::core::v3::HealthCheckFailureType type,
+                                  bool retriable);
```

Comment:
Compared to adding a boolean parameter, can we add a new HealthCheckFailure value `Retriable`?

Reply:
We could. That will also affect the `failure_type` displayed in the health event log (Unless we do some translation to make `RETRIABLE` -> `ACTIVE`). I had decided initially to go with the boolean to reduce the impact of this change, but we can certainly add a new type if desired.

Reply:
Interesting point. I think the boolean expresses the API nicely because the failure type is _still_ ACTIVE, it just happens to be retriable.